aws/resource_aws_sagemaker_model_test.go

```diff
@@ -0,0 +1,576 @@
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckSagemakerModelPrimaryContainerModelDataUrl(model *sagemaker.DescribeModelOutput, expected string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		modelDataUrl := model.PrimaryContainer.ModelDataUrl
+		if *modelDataUrl != expected {
+			return fmt.Errorf("Bad model data url of primary container: %s", *modelDataUrl)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckSagemakerModelPrimaryContainerHostname(model *sagemaker.DescribeModelOutput, expected string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		hostname := model.PrimaryContainer.ContainerHostname
+		if *hostname != expected {
+			return fmt.Errorf("Bad container hostname of primary container: %s", *hostname)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckSagemakerModelPrimaryContainerEnvironment(model *sagemaker.DescribeModelOutput, expected map[string]*string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		environment := model.PrimaryContainer.Environment
+
+		if !reflect.DeepEqual(environment, expected) {
+			return fmt.Errorf("Bad environment of primary container.")
+		}
+
+		return nil
+	}
+}
+
+const testAccSagemakerModelConfig = `
+resource "aws_sagemaker_model" "foo" {
+	name = "terraform-testacc-sagemaker-model-foo"
+	execution_role_arn = "${aws_iam_role.foo.arn}"
+
+	primary_container {
+		image = "174872318107.dkr.ecr.us-west-2.amazonaws.com/kmeans:1"
+	}
+}
+
+resource "aws_iam_role" "foo" {
+  name = "terraform-testacc-sagemaker-model-foo"
+  path = "/"
+  assume_role_policy = "${data.aws_iam_policy_document.assume_role.json}"
+}
+
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    actions = [ "sts:AssumeRole" ]
+    principals {
+      type = "Service"
+      identifiers = [ "sagemaker.amazonaws.com" ]
+    }
+  }
+}
+`
+
+const testAccSagemakerModelConfigTags = `
+resource "aws_sagemaker_model" "foo" {
+	name = "terraform-testacc-sagemaker-model-foo"
+	execution_role_arn = "${aws_iam_role.foo.arn}"
+
+	primary_container {
+		image = "174872318107.dkr.ecr.us-west-2.amazonaws.com/kmeans:1"
+	}
+
+	tags {
+		foo = "bar"
+	}
+}
+
+resource "aws_iam_role" "foo" {
+  name = "terraform-testacc-sagemaker-model-foo"
+  path = "/"
+  assume_role_policy = "${data.aws_iam_policy_document.assume_role.json}"
+}
+
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    actions = [ "sts:AssumeRole" ]
+    principals {
+      type = "Service"
+      identifiers = [ "sagemaker.amazonaws.com" ]
+    }
+  }
+}
+`
+
+const testAccSagemakerModelConfigTagsUpdate = `
+resource "aws_sagemaker_model" "foo" {
+	name = "terraform-testacc-sagemaker-model-foo"
+	execution_role_arn = "${aws_iam_role.foo.arn}"
+
+	primary_container {
+		image = "174872318107.dkr.ecr.us-west-2.amazonaws.com/kmeans:1"
+	}
+
+	tags {
+		bar = "baz"
+	}
+}
+
+resource "aws_iam_role" "foo" {
+  name = "terraform-testacc-sagemaker-model-foo"
```

Comment:
We also want to randomize this and parallelize these tests

Reply:
Done.